dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/Agent.java

```diff
@@ -0,0 +1,344 @@
+      try {
+        Thread.currentThread().setContextClassLoader(jmxFetchClassLoader);
+        final Class<?> jmxFetchAgentClass =
+            jmxFetchClassLoader.loadClass("datadog.trace.agent.jmxfetch.JMXFetch");
+        final Method jmxFetchInstallerMethod = jmxFetchAgentClass.getMethod("run");
+        jmxFetchInstallerMethod.invoke(null);
+        JMXFETCH_CLASSLOADER = jmxFetchClassLoader;
+      } finally {
+        Thread.currentThread().setContextClassLoader(contextLoader);
+      }
+    }
+  }
+
+  private static void configureLogger() {
+    setSystemPropertyDefault(SIMPLE_LOGGER_SHOW_DATE_TIME_PROPERTY, "true");
+    setSystemPropertyDefault(
+        SIMPLE_LOGGER_DATE_TIME_FORMAT_PROPERTY, SIMPLE_LOGGER_DATE_TIME_FORMAT_DEFAULT);
+
+    if (isDebugMode()) {
+      setSystemPropertyDefault(SIMPLE_LOGGER_DEFAULT_LOG_LEVEL_PROPERTY, "DEBUG");
+    }
+  }
+
+  private static void setSystemPropertyDefault(final String property, final String value) {
+    if (System.getProperty(property) == null) {
+      System.setProperty(property, value);
+    }
+  }
+
+  /**
+   * Create the datadog classloader. This must be called after the bootstrap jar has been appened to
+   * the bootstrap classpath.
+   *
+   * @param innerJarFilename Filename of internal jar to use for the classpath of the datadog
+   *     classloader
+   * @param bootstrapURL
+   * @return Datadog Classloader
+   */
+  private static ClassLoader createDatadogClassLoader(
+      final String innerJarFilename, final URL bootstrapURL) throws Exception {
+    final ClassLoader agentParent;
+    if (isJavaBefore9()) {
+      agentParent = null; // bootstrap
+    } else {
+      // platform classloader is parent of system in java 9+
+      agentParent = getPlatformClassLoader();
+    }
+
+    final Class<?> loaderClass =
+        ClassLoader.getSystemClassLoader().loadClass("datadog.trace.bootstrap.DatadogClassLoader");
+    final Constructor constructor =
+        loaderClass.getDeclaredConstructor(URL.class, String.class, ClassLoader.class);
+    return (ClassLoader) constructor.newInstance(bootstrapURL, innerJarFilename, agentParent);
+  }
+
+  private static ClassLoader getPlatformClassLoader()
+      throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+    /*
+     Must invoke ClassLoader.getPlatformClassLoader by reflection to remain
+     compatible with java 7 + 8.
+    */
+    final Method method = ClassLoader.class.getDeclaredMethod("getPlatformClassLoader");
+    return (ClassLoader) method.invoke(null);
+  }
+
+  /**
+   * Determine if we should log in debug level according to dd.trace.debug
+   *
+   * @return true if we should
+   */
+  private static boolean isDebugMode() {
+    final String tracerDebugLevelSysprop = "dd.trace.debug";
+    final String tracerDebugLevelProp = System.getProperty(tracerDebugLevelSysprop);
+
+    if (tracerDebugLevelProp != null) {
+      return Boolean.parseBoolean(tracerDebugLevelProp);
+    }
```

Comment:
Could be simplified by
```suggestion
    return Boolean.getBoolean(tracerDebugLevelSysprop);
```

Reply:
I do not think that would work: there is some logic below in this function so we do not want to exit unconditionally

Reply:
Ok. Np.